wrangler-transform/src/main/java/io/cdap/wrangler/Wrangler.java

```diff
@@ -524,6 +531,34 @@ private void validatePrecondition(String precondition, Boolean isConditionSQL, F
   }
 
   private void validateSQLModeDirectives(FailureCollector collector) {
+
+    String recipe = config.getDirectives();
+
+    registry = SystemDirectiveRegistry.INSTANCE;
```

Comment:
Also, this logic of loading directives is repeated in multiple places in this file. Make it a separate function?